content/post/postgres/extension-preloading.md

```diff
@@ -0,0 +1,170 @@
+    [`ALTER ROLE`], so it loads for every connection for that user:
+
+    ```sql
+    ALTER ROLE role_name
+      SET session_preload_libraries TO '$libdir/mylib';
+    ```
+
+*   Granting the user role the ability to set `session_preload_libraries`
+    which would allow them to use it (and any other shared library) in
+    `PGOPTIONS`:
+
+    ```sql
+    GRANT SET ON PARAMETER session_preload_libraries
+       TO role_name;
+    ```
+
+As an extension author, you don't need to do any special configuration, as
+long as your module is installed in the usual location via the [`MODULES`]
+`Makefile` variable. Still, it will be useful to document these options so
+that DBAs quickly see how to set things up for the users who need them.
+
+### Local Preloading
+
+As a special case, a DBA might want to make your debugging or
+performance-measurement extension available to any user who needs it, even
+unprivileged users. All they need to do is move it from `$libdir` to
+`$libdir/plugins`.
+
+Then any Postgres user can load it via the [`LOAD`] command or include it in
+their [`local_preload_libraries`] configuration via either [`PGOPTIONS`] or,
+for every connection, via `ALTER ROLE SET`:
+
+```sql
+ALTER ROLE role_name
+  SET local_preload_libraries TO '$libdir/plugins/mylib';
+```
+
+As an extension author, you don't need to do any special configuration; there
+is no `Makefile` variable to install it in the the `$libdir/plugins`
+directory. But it might be handy for DBAs to document this option *in addition
+to* the [session preloading](#session-preloading) options. But emphasize that
+it should be used if and *only if* they want to allow any and all of their
+users to load your extension library without any further intervention.
+
+### Shared Preloading
+
+The last preloading variable is [`shared_preload_libraries`], which should be
+required for modules to run in every session or to perform operations only
+available at service start up, such as [shared memory and lightweight locks]
+or starting [background workers].
+
+As an extension author, if your extension requires `shared_preload_libraries`
+preloading, the documentation should say so explicitly, and explain why. For
+examples of wording, see [pg_stat_statements], [sepgsql], and [auth_delay].
+
+Beyond these limited cases, any other modules can be added to
+[`shared_preload_libraries`] for efficiency purposes. However, since shared
+preload modules are loaded into every server process --- even if that process
+never uses the library --- preloading them is recommended only for libraries
+used in most sessions.
+
+As an extension author, it will be kind to DBAs to include this information,
+and to describe the circumstances under which they *might* want to preload
+your library in every service --- along with the caveat that doing so requires
+a server restart. For example wording, see [PL/Perl] and [auto_explain]
+
+### A Final Note
```

Comment:
Consider upgrading this to a top-level heading and naming it directly, something like "A Note on Dependencies"

Reply:
It applies only to preloaded libraries, under which the current header falls, but perhaps it's a more general issue given that it might just indicate a bug, as @turnstep suggests below.

Reply:
Converted to an h2 in efda14c.